tests/python_package_test/test_arrow.py

```diff
@@ -0,0 +1,102 @@
+# coding: utf-8
+import filecmp
+import tempfile
+from pathlib import Path
+from typing import Any, Callable, Dict
+
+import numpy as np
+import pyarrow as pa
+import pytest
+
+import lightgbm as lgb
+
```

Comment:
As written, this will cause something like `pytest tests/` to fail in an environment where `pyarrow` and `cffi` are not available.

Personally, I'm ok with keeping it that way. That is already how we handle `scikit-learn` tests (another example of tests that rely on a library that isn't a strict requirement for installing `lightgbm`): https://github.com/microsoft/LightGBM/blob/2d358d5d20e00837ea88455c953c921308c3b69e/tests/python_package_test/test_sklearn.py

I'd prefer that this fail with a big loud import error instead of risking a situation where these tests silently don't run in CI.

So not asking for anything to change, just calling that out here for the benefit of @jmoralez and other reviewers.

Reply:
The alternative would be to add something like this:

https://github.com/microsoft/LightGBM/blob/2d358d5d20e00837ea88455c953c921308c3b69e/tests/python_package_test/test_dask.py#L24-L25

I don't think we should.